12-kubernetes-cluster.tf

```diff
@@ -125,10 +125,24 @@ resource "azurerm_kubernetes_cluster" "kubernetes_cluster" {
       condition     = var.enable_automatic_channel_upgrade_patch != true || can(regex("^1\\.\\d\\d$", var.kubernetes_cluster_version))
       error_message = "When automatic upgrades are enabled, kubernetes_cluster_version must only include major and minor versions, not the patch version e.g. 1.18 or 1.25"
     }
+    precondition {
+      condition     = var.node_os_maintenance_window_duration != null && var.node_os_maintenance_window_duration > 3
+      error_message = "Duration must be at least 4 hours long"
+    }
   }
 
   automatic_channel_upgrade = var.enable_automatic_channel_upgrade_patch == true ? "patch" : null
   node_os_channel_upgrade   = var.enable_node_os_channel_upgrade_nodeimage == true ? "NodeImage" : null
+
+  maintenance_window_node_os {
+    duration    = var.node_os_maintenance_window_duration
+    frequency   = var.node_os_maintenance_window_frequency
+    day_of_week = var.node_os_maintenance_window_frequency != "Daily" ? var.node_os_maintenance_window_day_of_week : null
```

Comment:
need a second opinion on this one, if we lock down freq. to daily only we can remove `day_of_week` not sure to keep it in or not for now.

Reply:
How does this behave with everything defaulting to null?

Reply:
will need to plan out how it will behave, will also try and implement dynamic block to lock it down to sbox only.

Reply:
back to the drawing board, test failed in a different env so will have another look at it and get your thoughts